app/src/sharedTest/java/org/oppia/app/topic/TopicFragmentTest.kt

```diff
@@ -0,0 +1,342 @@
+      onView(withId(R.id.topic_name_text_view)).check(
+        matches(
+          withText(
+            containsString(topicName)
+          )
+        )
+      )
+    }
+  }
+
+  @Test
+  fun testTopicFragment_clickOnPlayTab_configurationChange_showsSameTabAndItsContent() {
+    launch(TopicActivity::class.java).use {
+      onView(
+        allOf(
+          withText(R.string.play),
+          isDescendantOfA(withId(R.id.topic_tabs_container))
+        )
+      ).perform(click())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.topic_tabs_container)).check(
+        matches(
+          matchCurrentTabTitle(
+            getResourceString(R.string.play)
+          )
+        )
+      )
+      onView(withText("First Story")).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testTopicFragment_clickOnTrainTab_configurationChange_showsSameTabAndItsContent() {
+    launch(TopicActivity::class.java).use {
+      onView(
+        allOf(
+          withText(R.string.train),
+          isDescendantOfA(withId(R.id.topic_tabs_container))
+        )
+      ).perform(click())
+      onView(withText("Master These Skills")).check(matches(isDisplayed()))
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.topic_tabs_container)).check(
+        matches(
+          matchCurrentTabTitle(
+            getResourceString(R.string.train)
+          )
+        )
+      )
+      onView(withText("Master These Skills")).check(matches(isDisplayed()))
+    }
+  }
+
+  @Test
+  fun testTopicFragment_clickOnReviewTab_configurationChange_showsSameTabAndItsContent() {
+    launch(TopicActivity::class.java).use {
+      onView(
+        allOf(
+          withText(R.string.review),
+          isDescendantOfA(withId(R.id.topic_tabs_container))
+        )
+      ).perform(click())
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.topic_tabs_container)).check(
+        matches(
+          matchCurrentTabTitle(
+            getResourceString(R.string.review)
+          )
+        )
+      )
+      onView(
+        atPositionOnView(R.id.review_skill_recycler_view, 0, R.id.skill_name)
+      ).check(matches(withText(containsString("An important skill"))))
+    }
+  }
+
+  @Test
+  fun testTopicFragment_configurationChange_showsDefaultTabAndItsContent() {
+    launch(TopicActivity::class.java).use {
+      onView(isRoot()).perform(orientationLandscape())
+      onView(withId(R.id.topic_tabs_container)).check(
+        matches(
+          matchCurrentTabTitle(
+            getResourceString(R.string.overview)
+          )
+        )
+      )
+      onView(withId(R.id.topic_name_text_view)).check(
+        matches(
+          withText(
+            containsString(topicName)
+          )
+        )
+      )
+    }
+  }
+
+  private fun getResourceString(id: Int): String {
+    val resources =
+      InstrumentationRegistry.getInstrumentation().targetContext.resources
```

Comment:
I believe this won't work correctly when run on Robolectric, and I think we can rely on the AndroidX application provider to get access to resources:

```kotlin
val resources = ApplicationProvider.getApplicationContext().resources
```

Reply:
This is giving an error 
`/Users/marysoloman/StudioProjects/oppia-android/app/build/tmp/kapt3/stubs/debug/org/oppia/app/application/ApplicationComponent.java:11: error: [Dagger/MissingBinding] org.oppia.app.home.UserAppHistoryViewModel cannot be provided without an @Inject constructor or an @Provides-annotated method.
public abstract interface ApplicationComponent {
                ^
      javax.inject.Provider<org.oppia.app.home.UserAppHistoryViewModel> is injected at
          org.oppia.app.viewmodel.ViewModelBridgeFactory(viewModelProvider)
      org.oppia.app.viewmodel.ViewModelBridgeFactory<org.oppia.app.home.UserAppHistoryViewModel> is injected at
          org.oppia.app.viewmodel.ViewModelProvider(bridgeFactory)
      org.oppia.app.viewmodel.ViewModelProvider<org.oppia.app.home.UserAppHistoryViewModel> is injected at
          org.oppia.app.home.HomeFragmentPresenter(…, viewModelProvider, …)
      org.oppia.app.home.HomeFragmentPresenter is injected at
          org.oppia.app.home.HomeFragment.homeFragmentPresenter
      org.oppia.app.home.HomeFragment is injected at
          org.oppia.app.fragment.FragmentComponent.inject(org.oppia.app.home.HomeFragment) [org.oppia.app.application.ApplicationComponent → org.oppia.app.activity.ActivityComponent → org.oppia.app.fragment.FragmentComponent]
`